pkg/controller/v1beta1/inferenceservice/reconcilers/deployment/deployment_reconciler.go

```diff
@@ -124,12 +130,45 @@ func createRawDeployment(componentMeta metav1.ObjectMeta, workerComponentMeta me
 	return deploymentList
 }
 
-func createRawDefaultDeployment(componentMeta metav1.ObjectMeta,
+func createRawDefaultDeployment(clientset kubernetes.Interface, componentMeta metav1.ObjectMeta,
 	componentExt *v1beta1.ComponentExtensionSpec,
 	podSpec *corev1.PodSpec) *appsv1.Deployment {
 	podMetadata := componentMeta
 	podMetadata.Labels["app"] = constants.GetRawServiceLabel(componentMeta.Name)
 	setDefaultPodSpec(podSpec)
+	var isvcname string
+	if val, ok := componentMeta.Labels[constants.InferenceServiceLabel]; ok {
+		isvcname = val
+	} else {
+		isvcname = componentMeta.Name
+	}
+	var upstreamPort string
+	if val, ok := componentMeta.Labels[constants.ODHKserveRawAuth]; ok && val == "true" {
+		if componentExt != nil && componentExt.Batcher != nil {
+			upstreamPort = constants.InferenceServiceDefaultAgentPortStr
+		} else if componentExt != nil && componentExt.Logger != nil {
+			upstreamPort = constants.InferenceServiceDefaultAgentPortStr
+		} else {
+			upstreamPort = GetKServeContainerPort(podSpec)
+			if upstreamPort == "" {
+				upstreamPort = constants.InferenceServiceDefaultHttpPort
+			}
+		}
+		oauthProxyContainer, err := generateOauthProxyContainer(clientset, isvcname, componentMeta.Namespace, upstreamPort)
+		if err == nil {
```

Comment:
the  `generateOauthProxyContainer`func will always return `nil` for error, maybe you could improve this error handling.

Reply:
Sorry, butignoring errors is not tolerable for me.

You already changed the function signature to add the `clientset` parameter. So, you already had to change all callers.

The usages of this func is just this deployment_reconciler file, and tests: 
![image](https://github.com/user-attachments/assets/40daa8ff-ed3d-4228-821d-c2e754ced275)

In turn, the `NewDeploymentReconciler` func has only one usage:
![image](https://github.com/user-attachments/assets/e7c8dd33-2fa3-40bb-8a29-6087ee9bc5f3)

The `NewRawKubeReconciler` does have `error` as return value, so the "refactoring" would end there. I'm not seeing a major change.

Reply:
modified to return err